utils/utils.go

```diff
@@ -112,6 +115,36 @@ func TouchCopy(writer io.Writer, reader io.Reader) error {
 	return nil
 }
 
+// TouchCopyBuffer is a wrapper around io.CopyBuffer which ensures that even empty source files (reader) will get written as an
+// empty file. It guarantees a Write() call on the target file.
+// bufferSize is in bytes and if is less than TouchCopyMinBufferSize will result in a buffer of size TouchCopyMinBufferSize
+// bytes. If bufferSize is > TouchCopyMinBufferSize it will result in a buffer of size bufferSize bytes
+func TouchCopyBuffered(writer io.Writer, reader io.Reader, bufferSize int) error {
+	var buffer []byte
+	var size int64
+	var err error
+
+	if bufferSize > TouchCopyMinBufferSize {
```

Comment:
Most of the optimization is here such that it impacts existing code using functions that make use of TouchCopy previously.